### PR TITLE
Mailto - Spam Bot improvement - thanks @waschier-design

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Encode E-Mail Adress and make use of Smarty mailto & encode function for better spam bot protection. Pick https://github.com/PrestaShop/PrestaShop/pull/9180 that was abandoned
| Type?         | improvement
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | Please check in the footer, the contact mail works as before. Now it's protected against SPAM bots. See screenshot.

Capture d’écran 2020-02-07 à 16 18 17

Pick https://github.com/PrestaShop/PrestaShop/pull/9180 that was abandoned

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17562)

### CODE REVIEW

**Comment by Progi1984** [comment]
Travis is now :green_heart:

File: themes/classic/modules/ps_contactinfo/ps_contactinfo.tpl [hunk 1]
```diff
@@ -54,16 +54,8 @@
...
-          s='Email us: [1]%email%[/1]'
-          sprintf=[
-            '[1]' => '<a href="mailto:'|cat:$contact_infos.email|cat:'" class="dropdown">',
-            '[/1]' => '</a>',
-            '%email%' => $contact_infos.email
-          ]
-          d='Shop.Theme.Global'
-        }
+          {l s='Email us:' d='Shop.Theme.Global'}
+          {mailto address={$contact_infos.email} encode="javascript"}
```
Comment (Progi1984) [inline themes/classic/modules/ps_contactinfo/ps_contactinfo.tpl]:
```suggestion
          {mailto address=$contact_infos.email encode="javascript"}
```

File: themes/classic/modules/ps_contactinfo/ps_contactinfo-rich.tpl [hunk 2]
```diff
@@ -56,7 +56,7 @@
       <div class="data email">
         {l s='Email us:' d='Shop.Theme.Global'}<br/>
       </div>
-      <a href="mailto:{$contact_infos.email}">{$contact_infos.email}</a>
+      {mailto address={$contact_infos.email} encode="javascript"}
```
Comment (Progi1984) [inline themes/classic/modules/ps_contactinfo/ps_contactinfo-rich.tpl]:
```suggestion
      {mailto address=$contact_infos.email encode="javascript"}
```

**Comment by matks** [comment]
> Travis is now 💚

Did you do something ?

**Comment by Progi1984** [comment]
@matks I re-launch two jobs ;)

**Review by PierreRambaud — APPROVED** [review]

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
See @Progi1984's comment

File: themes/classic/modules/ps_contactinfo/ps_contactinfo-rich.tpl (hunk 2 above)
Comment (matks) [reply themes/classic/modules/ps_contactinfo/ps_contactinfo-rich.tpl]: Fixed in https://github.com/PrestaShop/PrestaShop/pull/17562/commits/d5f5be793c60dd6219fa5977f9595289dfa4063c

File: themes/classic/modules/ps_contactinfo/ps_contactinfo.tpl (hunk 1 above)
Comment (matks) [reply themes/classic/modules/ps_contactinfo/ps_contactinfo.tpl]: Fixed in https://github.com/PrestaShop/PrestaShop/pull/17562/commits/d5f5be793c60dd6219fa5977f9595289dfa4063c

File: themes/classic/modules/ps_contactinfo/ps_contactinfo.tpl [hunk 3]
```diff
@@ -54,21 +54,14 @@
...
-          d='Shop.Theme.Global'
-        }
+          {l s='Email us:' d='Shop.Theme.Global'}
+          {mailto address=$contact_infos.email encode="javascript"}
       {/if}
   </div>
   <div class="hidden-md-up">
     <div class="title">
       <a class="h3" href="{$urls.pages.stores}">{l s='Store information' d='Shop.Theme.Global'}</a>
+
```
Comment (Progi1984) [inline themes/classic/modules/ps_contactinfo/ps_contactinfo.tpl:64]: Remove this added empty line. Thanks

Comment (matks) [reply themes/classic/modules/ps_contactinfo/ps_contactinfo.tpl:64]: But I like it like this 🎶

**Review by atomiix — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by SD1982** [comment]
Mailto work like before as expected. Thanks @matks

**Comment by matks** [comment]
Well, thanks @waschier-design who is the original author 😄

**Pull request merged**